src/main/java/net/oauth/jsontoken/exceptions/ErrorCode.java

```diff
@@ -0,0 +1,45 @@
+package net.oauth.jsontoken.exceptions;
+
+public enum ErrorCode {
+ /**
+  * Signature failed verification.
+  */
+ BAD_SIGNATURE,
+
+ /**
+  * IAT is after EXP or IAT is in the future
+  */
+ BAD_TIME_RANGE,
+
+ /**
+  * IAT and EXP are both in the past.
+  */
+ EXPIRED_TOKEN,
+
+ /**
+  * Something went wrong internally.
+  */
+ INTERNAL_ERROR,
+
+ /**
+  * Token string is corrupted and/or does not contain three components.
+  */
+ MALFORMED_TOKEN_STRING,
+
+ /**
+  * There are no verifiers available for a given issuer.
+  */
+ NO_VERIFIER,
+
+ /**
+  * Generic catch-all for unknown and ambiguous exceptions.
+  */
+ UNKNOWN,
```

Comment:
What are the existing use cases for this? How does it differ from INTERNAL_ERROR?

Reply:
So far, the use case I intended for this is to catch exceptions where it is somewhat expected (eg. when one of the checkers failed), but we don't have a definite criteria of what failed. INTERNAL_ERROR would be for cases when the program is in an unexpected place. (eg. using the JsonToken's parsing constructor to construct a token, then using that to sign something).

The description and/or enum name might need to change to better reflect what it's supposed to represent. Alternatively, we could just get rid of INTERNAL_ERROR altogether and just directly throw these errors without the custom exception wrapping around it.

Reply:
I have opted to remove INTERNAL_ERROR and just include it under the UNKNOWN error code. Additionally, I have changed the description of UNKNOWN to better reflect what will be thrown (not predefined errors).